config/config.go

```diff
@@ -446,6 +446,10 @@ type Debug struct {
 	TimeoutNotification TimeoutNotification `mapstructure:"timeout_notification"`
 }
 
+type DebugInfo struct {
+	Allow bool `yaml:"allow" json:"allow"`
```

Comment:
The default in Go for a boolean is false, so that will result in every adapter being disallowed unless all the BidderInfo yamls are updated to allow this.

Reply:
@SyntaxNode please advice if we want to have` DebugRestricted` instead of `DebugInfo.Allow`

Reply:
I think you missed my point. I was thinking of moving this flag out of the `BidderInfo` struct, and thus out of the bidder/info endpoint, and into the actual server configuration in the `Adapter` struct, where we actually do sever level configuration of the adapter. I am not aware of anything in the /static/bidder-info/ directory that a PBS host should touch, all configuration should be driven by the config file and/or environment variables. Spreading configuration out across multiple files will just lead to confusion.

Reply:
I'm confused. 
We want to have bidder level debug flag and store it in static/bidder-info/{bidder}.yaml files. The only place where we load it is in adapters/info.go -> func ParseBidderInfos. We Unmarshal file data to BidderInfo struct. 
Unless I miss something... 
Do you mean I need to  move it to  config/adapter.go -> Adapter struct after loading? 
What is "the actual server configuration in the Adapter struct"?

Reply:
> @SyntaxNode please advice if we want to have DebugRestricted instead of DebugInfo.Allow

I don't have a strong opinion. Bret introduced this config as debug allow and that makes sense to me, but yes - it does play against the boolean zero value. We'd need to ask in the GitHub issues to swap it to something like restricted.

> I was thinking of moving this flag out of the BidderInfo struct, and thus out of the bidder/info endpoint, and into the actual server configuration in the Adapter struct,

I think both have valid arguments. I had spec'd it for the bidder yaml since I didn't expect a host to override this value. I thought if the bidder knows its url will include sensitive information, then it could specify such here for all hosts to follow.

We've also had discussions about in the future using bidder yaml as the base config for each adapter as is done in PBS-Java and then using the config system to override those values if needed. This would work well for reducing the code length of config.go, making it easier to port adapters to PBS-Java, and would provide a nice way to override yaml information for aliases.

Reply:
Oh, if PBS-Java is already using it for config, then it is not such a big change to for us to do the same. I have no opinion on if this struct should be in config/config.go or adapters/info.go